overlord/devicestate/firstboot_preseed_test.go

```diff
@@ -0,0 +1,224 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015-2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package devicestate_test
+
+import (
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/bootloader/bootloadertest"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/seed/seedtest"
+	"github.com/snapcore/snapd/testutil"
+)
+
+type firstbootPreseed16Suite struct {
+	firstBootBaseTest
+	firstBoot16BaseTest
+}
+
+var _ = Suite(&firstbootPreseed16Suite{})
+
+func checkPreseedTasks(c *C, tsAll []*state.TaskSet) {
+	// the tasks of the last taskset must be mark-preseeded, mark-seeded
```

Comment:
minor nitpick, maybe add something about the ordering being specific?

```suggestion
	// the tasks of the last taskset must be mark-preseeded, mark-seeded, in that order
```